### PR TITLE
Generate Manifests alongside build artifacts

### PR DESCRIPTION
**Pull request by scbedd**
This adds two steps to the `build-artifacts` stage of our yaml pipelines.

Related to Azure/azure-sdk-tools#2273

### CODE REVIEW

File: eng/pipelines/templates/steps/build-artifacts.yml [hunk 1]
```diff
@@ -101,6 +101,17 @@ steps:
 
   - ${{ parameters.BeforePublishSteps }}
 
+  - task: AzureArtifacts.manifest-generator-task.manifest-generator-task.ManifestGeneratorTask@0
+    displayName: 'Upload Package SBOM'
+    inputs:
+      BuildDropPath: $(Build.ArtifactStagingDirectory)
+
+  - task: AzureArtifacts.manifest-generator-task.manifest-generator-task.ManifestGeneratorTask@0
```
Comment (weshaggard) [inline eng/pipelines/templates/steps/build-artifacts.yml:109]: Do we need a sbom for the docs?

File: eng/pipelines/templates/variables/globals.yml [hunk 2]
```diff
@@ -1,4 +1,5 @@
 variables:
   PythonVersion: '3.6'
   skipComponentGovernanceDetection: true
-  AzureSDKCondaChannel: https://azuresdkconda.blob.core.windows.net/channel1/
\ No newline at end of file
+  AzureSDKCondaChannel: https://azuresdkconda.blob.core.windows.net/channel1/
+  Package.EnableSBOMSigning: true
```
Comment (weshaggard) [inline eng/pipelines/templates/variables/globals.yml:5]: Do we need to set this yet? Will it even work for public builds?

**Review by weshaggard — APPROVED** [review]

File: eng/pipelines/templates/variables/globals.yml (hunk 2 above)
Comment (scbedd) [reply eng/pipelines/templates/variables/globals.yml:5]: Either way, it won't do any harm! (At least not according to their documentation)

I think that at the outset they won't enforce on this, but they _will_. Leaving it on by default is good imo.

File: eng/pipelines/templates/steps/build-artifacts.yml (hunk 1 above)
Comment (scbedd) [reply eng/pipelines/templates/steps/build-artifacts.yml:109]: I think so, yes. I figure it's a shipped artifact so 🤷‍♂️

**Pull request merged**